pkg/restore/restore.go

```diff
@@ -1507,6 +1507,22 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		}
 	}
 
+	// Check if cluster has GVK required to restore resource, otherwise return err.
+	// Placed here because the object apiVersion might get modified by a RestorePlugin
+	// So we check for the GVK after the RestorePlugin has run.
+	_, _, err = ctx.discoveryHelper.KindFor(schema.GroupVersionKind{
```

Comment:
I'm afraid this change isn't the correct fix because the informer syncing will still stuck if the CRDs don't exist or aren't restored: https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L602

The second commit is the good one

Reply:
Thanks. So maybe hybrid of both?

Reply:
IMHO, the change in the second commit is enough. Seems the change here doesn't help resovling the issue, correct me if I'm wrong.